numpy/core/src/umath/ufunc_object.c

```diff
@@ -1553,12 +1553,11 @@ execute_fancy_ufunc_loop(PyUFuncObject *ufunc,
         }
 
         /*
-         * Put the updated operand back and undo the DECREF above. If
-         * COPY_IF_OVERLAP made a temporary copy, the output will be copied
-         * by UPDATEIFCOPY even if op[i] was changed by prepare_ufunc_output.
+         * Put the updated operand back.  If COPY_IF_OVERLAP made a temporary
+         * copy, the output will be copied by UPDATEIFCOPY even if op[i]
+         * was changed by prepare_ufunc_output.
          */
-        op[i] = op_tmp;
-        Py_DECREF(op_tmp);
+        op[i+nin] = op_tmp;
```

Comment:
It is a bit annoying, but just to note that the whole `execute_fancy_ufunc_loop` is deleted in the follow-up.